core/create/src/prompts/options.ts

```diff
@@ -1,126 +1,146 @@
 import { rootLogger as winstonLogger, styles } from '@dotcom-tool-kit/logger'
 import type { RCFile } from '@dotcom-tool-kit/types'
-import type { Schema, SchemaPromptGenerator, SchemaType } from '@dotcom-tool-kit/types/src/schema'
+import type { PromptGenerators } from '@dotcom-tool-kit/types/src/schema'
 import type { ValidConfig } from 'dotcom-tool-kit/lib/config'
 import { promises as fs } from 'fs'
 import * as yaml from 'js-yaml'
 import partition from 'lodash/partition'
 import prompt from 'prompts'
+import { z } from 'zod'
 import type { Logger } from '../logger'
 
+interface OptionSettings {
+  name: string
+  type: z.ZodTypeAny
+  default?: any
+}
+
 async function optionsPromptForPlugin(
   toolKitConfig: RCFile,
   plugin: string,
-  options: [string, SchemaType][]
+  options: OptionSettings[]
 ): Promise<boolean> {
   let pluginCancelled = false
   const onCancel = () => {
     pluginCancelled = true
   }
 
-  for (const [optionName, optionType] of options) {
-    switch (optionType) {
-      case 'string':
+  for (const { name: optionName, type: optionType, default: optionDefault } of options) {
+    const defaultSuffix = optionDefault
+      ? ` (leave blank to use default value ${styles.code(JSON.stringify(optionDefault))})`
+      : ''
+
+    switch (optionType._def.typeName as z.ZodFirstPartyTypeKind) {
```

Comment:
**suggestion:** before this change this line was `switch (optionType)`, and now it is `switch (optionType._def.typeName as z.ZodFirstPartyTypeKind)` which is less readable and understandable. Could this be made easier to read with descriptive variable names (or some other solution)?

Reply:
I'll split it out into a variable assignment and add a comment explaining why we're using the `_def` field – does that sound better?

Reply:
Yep!